model/task_lifecycle.go

```diff
@@ -30,35 +30,29 @@ type StatusChanges struct {
 	BuildComplete    bool
 }
 
-func SetActiveState(taskId string, caller string, active bool) error {
-	t, err := task.FindOne(task.ById(taskId))
-	if err != nil {
-		return err
-	}
-	if t == nil {
-		return errors.Errorf("task '%s' not found", taskId)
-	}
+func SetActiveState(t *task.Task, caller string, active bool) error {
 	if active {
 		// if the task is being activated and it doesn't override its dependencies
 		// activate the task's dependencies as well
+		tasksToActivate := []task.Task{}
 		if !t.OverrideDependencies {
-			for _, dep := range t.DependsOn {
-				if err = SetActiveState(dep.TaskId, caller, true); err != nil {
-					return errors.Wrapf(err, "error activating dependency for %v with id %v",
-						taskId, dep.TaskId)
-				}
+			deps, err := task.GetRecursiveDependenciesUp([]task.Task{*t}, nil)
+			if err != nil {
+				return errors.Wrapf(err, "error getting tasks '%s' depends on", t.Id)
 			}
+			tasksToActivate = append(tasksToActivate, deps...)
 		}
 
 		if t.DispatchTime != utility.ZeroTime && t.Status == evergreen.TaskUndispatched {
-			if err = resetTask(t.Id, caller); err != nil {
+			if err := resetTask(t.Id, caller); err != nil {
 				return errors.Wrap(err, "error resetting task")
 			}
 		} else {
-			if err = t.ActivateTask(caller); err != nil {
-				return errors.Wrap(err, "error while activating task")
-			}
-			event.LogTaskActivated(taskId, t.Execution, caller)
```

Comment:
Why do we remove these logs?

Reply:
The event is logged at the end of `task.ActivateTasks`. 
Consolidating logging the event into the function that makes the event makes sure developers don't forget to log the event.